testing/checkers.go

```diff
@@ -138,19 +190,44 @@ func (checker *dirEqualsChecker) Check(params []interface{}, names []string) (re
 
 		data, err := ioutil.ReadFile(filepath.Join(dirPath, f.Name()))
 		if err != nil {
-			return false, fmt.Sprintf("failed to read file '%s' from directory: %s", f.Name(), err.Error())
+			return nil, fmt.Errorf("failed to read file '%s' from directory: %s", f.Name(), err.Error())
 		}
 		obtained[f.Name()] = string(data)
 	}
+	return obtained, nil
+}
 
-	isOk := reflect.DeepEqual(obtained, params[1])
-
-	// When match is false, we show user the content, not the filepath.
-	if !isOk {
-		params[0] = obtained
+func compareMaps(obtained map[string]string, expected map[string]interface{}) error {
+	if len(obtained) != len(expected) {
+		return fmt.Errorf("obtained map key is not as expected")
 	}
 
-	return isOk, ""
+	for key, val := range expected {
+		_, exists := obtained[key]
+		if !exists {
+			return fmt.Errorf("expected key '%s' not found in obtained map", key)
+		}
+
+		if expectedString, ok := val.(string); ok {
+			if expectedString != obtained[key] {
+				return fmt.Errorf("mismatch for key '%s': '%s' != '%s'", key, val, obtained[key])
+			}
+			continue
```

Comment:
Is there a special meaning to these `continue`? They are always at the end of their corresponding `if/else` and after this `if-else` there is nothing you would like to skip. I suggest removing it.